pyrec/implicit/als.py

```diff
@@ -142,6 +148,15 @@ def load_recommender(als_model_file: str, index_file: str, item_feature_file: st
             from .annoy_item_features import ImplicitAnnoyItemFeatureRecommender
             return ImplicitAnnoyItemFeatureRecommender(model, recommend_index, max_norm, user_labels,item_labels,
                                                        tag_tfidf_transformer, tag_lookup, item_embedding_weight)
-
+    elif index_file.endswith('.hnsw'):
+        import hnswlib
+        from .hnsw import ImplicitHNSWRecommender
+        log.info("Loading hnsw recommendation index")
+        # we build the index in l2 space and load it in inner product space on purpose.
+        # This space change gives us 0.96 recall
+        l2_recommend_index = hnswlib.Index(space='ip', dim=model.item_factors.shape[1])
+        l2_recommend_index.load_index(index_file)
+        l2_recommend_index.set_ef(2000)
```

Comment:
should that `2000`  be parametrisable?
you could add `**kwargs` to the method

Reply:
Or should the caller then adjust the value?

Reply:
yes,  2000 can be parametrisable, I am just thinking set the ef once should be enough, not sure how many over header will set ef spend for every query.